docs/pages/guides/schema-validation/language.mdx

```diff
@@ -0,0 +1,263 @@
+---
+meta:
+  title: "Schema language"
+  parentTitle: "Guides"
+  description: "Liveblocks schema language specification"
+---
+
+<Banner title="Private beta">
+
+Schema validation is currently in private beta. If you would like to try it,
+please fill out the following survey to join our
+[waiting list](https://docs.google.com/forms/d/197IrpiyeqlJo9xOCOfzwdcwBIk6nisk9pToJe3st3Ng/viewform).
+
+</Banner>
+
+## Syntax
+
+The list below is exhaustive of the supported feature that will be implemented
+before the public beta.
+
+We currently support the following schema types: `Scalars`, `Object` and
+`LiveObject` during the private beta. We’re sharing our plans for other syntaxes
+so you can give us early feedback [here](TODO).
+
+### Storage root
+
+Each schema must include the `Storage` type, a special type of “root” object.
+
+```ts
+type Storage {
+
+}
+```
+
+### Scalars
+
+Familiar scalar types are globally available when you create a schema. We
+support:
+
+- `String`
+- `Int` (only whole numbers)
+- `Float` (floats and whole numbers)
+- `Boolean`
+
+A sample schema using scalar types could look like this:
+
+```ts
+type Storage {
+  name: String
+  age: Int
+  height: Float
+  hasSiblings: Boolean
+}
+```
+
+And here are some updates that would accepted and rejected by the schema:
+
+```ts
+// ✅ Valid storage updates
+root.set("name", "Marie Curie");
+root.set("age", 66);
+root.set("height", 1.52);
+root.set("hasSiblings", true);
+
+// ❌ Invalid storage updates
+root.set("name", true);
+root.set("age", 66.5); // Invalid because `age` should be a whole number
+root.set("height", "1.52");
+root.set("hasSiblings", "???");
+```
+
+### Optionals
+
+Each field inside an object type can be marked optional using the `?` operator.
+An optional field means that it can be deleted.
+
+For example, if your schema looked exactly like the example provided above,
+except we changed the `age` field to optional:
+
+```ts highlight="3"
+type Storage {
+  name: String
+  age?: Int
```

Comment:
love the sequence you did here with age being detailed above, and now it's optional. Brilliant 🙇